src/runtime/manifest-ast-nodes.ts

```diff
@@ -87,6 +87,7 @@ export interface TypeVariable extends BaseNode {
   kind: 'variable-type';
   name: string;
   constraint: ParticleHandleConnection;
+  resolveToMaxType: boolean;
```

Comment:
Might be good to have a comment about the purpose of this (i.e. that it allows a type variable to act as a 'catch all'... I'm still unsure about this approach but I think this implementation makes sense (given the approach)

Reply:
After considering everything in context, I realized that I didn't need to process this at the parser level -- I could move the logic to the analyzer. So, I did. I think this is a cleaner approach. WDYT?